turbo/snapshotsync/freezeblocks/beacon_block_reader.go

```diff
@@ -45,48 +46,81 @@ func (r *beaconSnapshotReader) FrozenSlots() uint64 {
 }
 
 func (r *beaconSnapshotReader) ReadBlock(slot uint64) (*cltypes.SignedBeaconBlock, error) {
-	buf, err := r.RawBlockSSZ(slot)
-	if err != nil {
-		return nil, err
+	view := r.sn.View()
+	defer view.Close()
+
+	var buf []byte
+
+	seg, ok := view.BeaconBlocksSegment(slot)
+	if !ok {
+		return nil, nil
 	}
-	if buf == nil {
+
+	if seg.idxSlot == nil {
 		return nil, nil
 	}
+	blockOffset := seg.idxSlot.OrdinalLookup(slot - seg.idxSlot.BaseDataID())
 
-	// Use pooled buffers and readers to avoid allocations.
+	gg := seg.seg.MakeGetter()
+	gg.Reset(blockOffset)
+	if !gg.HasNext() {
+		return nil, nil
+	}
+
+	buf, _ = gg.Next(buf)
```

Comment:
buf[:0]

Reply:
buf is `var buf []byte` no need to do that

Reply:
Yes, but It’s fragile. 
If i copy this pattern in another place it may become bug. 
or if in future i will change code to
for { 
   buf, _ = g.Next(buf)
}
It will become a bug

Reply:
FYI: also can do Next(nil)